my_game.py

```diff
@@ -90,6 +101,90 @@ def update(self):
             self.kill()
 
 
+class UFOShot(arcade.Sprite):
+    """shot fired by the ufo"""
+
+    def __int__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def update(self):
+        """update position/kill if out of bounds"""
+
+        self.center_x += self.change_x
+        self.center_y += self.change_y
+
+        self.angle = ((self.change_x / self.change_y) * -90)  # set ange based on our direction
+
+        # kill if out of bounds
+        if self.center_x > SCREEN_WIDTH or self.center_x < 0 and self.center_y > SCREEN_HEIGHT or self.center_y < 0:
+            self.kill()
+
+
+class BonusUFO(arcade.Sprite):
+    """occasionally moves across the screen. Grants the player points if shot"""
+
+    def __int__(self, shot_list, **kwargs):
+
+        kwargs['filename'] = "images/ufoBlue.png"
+
+        kwargs['scale'] = SPRITE_SCALING
+
+        # set random position off-screen
+        kwargs['center_x'] = random.choice([0, SCREEN_WIDTH])
+        kwargs['center_y'] = random.choice([0, SCREEN_HEIGHT])
+
+        # send arguments upstairs
+        super().__init__(**kwargs)
+
+        self.shot_list = shot_list
+
+        # set random direction. always point towards center, with noise
+        self.change_x = random.randrange(1, UFO_SPEED)
+        if self.center_x > SCREEN_WIDTH / 2:
+            self.change_x *= -1
+
+        self.change_y = UFO_SPEED - self.change_x
+        if self.center_y > SCREEN_HEIGHT / 2:
+            self.change_y *= -1
+
+        # setup direction changing
+        def change_dir(delta_time):
+            r = random.randrange(-UFO_SPEED, UFO_SPEED)
+            self.change_x -= r
+            self.change_y += r
+
+        arcade.schedule(change_dir, UFO_DIR_CHANGE_RATE)
+
+        # setup shooting
+        arcade.schedule(self.shoot, UFO_FIRE_RATE)
+
+    def shoot(self, delta_time):
+        new_ufo_shot = UFOShot()  # sprites created with arcade.schedule don't __init__ it has to be manually called
+        new_ufo_shot.__init__(
+            filename="images/Lasers/laserGreen07.png",
```

Comment:
Let's put the path in the class UFOShot.

Reply:
yes, but it seems like when i do just that it doesn't set filename. Like this
`kwargs['filename'] = "images/Lasers/laserGreen07.png"`

that line seems to not execute while it is in BonusUFO.__init__... (same thing for a print statement)

Reply:
I'll allow it. We can talk about those issues and try to figure them out later :+1:

Reply:
I'll allow it. We can talk about those issues and try to figure them out later :+1: